wgengine/tstun/tun.go

```diff
@@ -267,16 +316,40 @@ func (t *TUN) SetFilter(filt *filter.Filter) {
 // InjectInbound makes the TUN device behave as if a packet
 // with the given contents was received from the network.
 // It blocks and does not take ownership of the packet.
-// Injecting an empty packet is a no-op.
-func (t *TUN) InjectInbound(packet []byte) error {
+//
+// The packet contents are to start at &buf[offset].
+// If the underlying TUN device is real,
+// offset must be greater or equal to PacketStartOffset.
```

Comment:
What happens if I violate this constraint? Does it fail safe (e.g. log unexpected and drop the packet), or will it panic later?

Reply:
It's slightly difficult to check since one would have to know whether the underlying device is real (none of `fakeTUN`, `chTUN`, etc) before erroring out justifiably. Perhaps it's best to always error out if `offset < PacketStartOffset` and change all the fake-mode code to use `InjectInboundCopy` or proper offsets.

Reply:
It would probably be best if the `Tun` had a method to get at this minimal offset, since it's not at all an implementation detail for performance-conscious callers.